builtin/providers/aws/resource_aws_elb.go

```diff
@@ -438,6 +468,18 @@ func resourceAwsElbUpdate(d *schema.ResourceData, meta interface{}) error {
 			},
 		}
 
+		logs := d.Get("access_logs").(*schema.Set).List()
+		if len(logs) > 0 {
+			log := logs[0].(map[string]interface{})
+			accessLogs := &elb.AccessLog{
+				Enabled:      aws.Bool(log["enabled"].(bool)),
+				EmitInterval: aws.Int64(log["interval"].(int64)),
+				S3BucketName: aws.String(log["bucket"].(string)),
+				S3BucketPrefix: aws.String(log["bucket"].(string)),
+			}
```

Comment:
For any of the above `log` attributes that are optional, omitting them will return their default value here (strings will be `""`), and I suspect the API will be unhappy with this.

for any optional attribute, we likely need to check that they exist and only add them if they do:

In the code below, when constructing `&elb.AccessLog`, we should test to see if it's empty and only add it if not. 

Something like so:

```go
accessLogs := &elb.AccessLog{
  Enabled:      aws.Bool(log["enabled"].(bool)),
  EmitInterval: aws.Int64(log["interval"].(int64)),
}

if log["bucket"] != "" {
  accessLogs.S3BucketPrefix = aws.String(log["bucket"].(string))
}

// same for bucket name
```

`EmitInterval` is likely OK, since it's default is the AWS default.